### PR TITLE
WOR-649 spring 3.0, etc.

### PR DESCRIPTION
**Pull request by dvoet**
https://broadworkbench.atlassian.net/browse/WOR-649

upgrade to spring boot 3 and jakarta
switch from opencensus to opentelemetry

### CODE REVIEW

**Comment by cahrens** [comment]
Can you also remove the override in dependabot  configuration? https://github.com/DataBiosphere/terra-billing-profile-manager/pull/95/files

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ dependencies {
     implementation 'com.github.spotbugs.snom:spotbugs-gradle-plugin:5.1.4'
     implementation 'io.spring.dependency-management:io.spring.dependency-management.gradle.plugin:1.1.3'
     implementation 'org.hidetake.swagger.generator:org.hidetake.swagger.generator.gradle.plugin:2.19.2'
-    implementation 'org.springframework.boot:spring-boot-gradle-plugin:2.7.17'
+    implementation 'org.springframework.boot:spring-boot-gradle-plugin:3.1.2'
     implementation 'bio.terra:terra-test-runner:0.2.0-SNAPSHOT'
```
Comment (cahrens) [inline buildSrc/build.gradle:20]: I wonder if we can remove this override/direct dependency now.

https://github.com/DataBiosphere/terra-billing-profile-manager/blob/59bceb86b4f44d38bf71c12e30f6e698d0e716e1/service/build.gradle#L18

**Comment by cahrens** [comment]
@blakery I can't figure out how to get to the artifact that explains which pact tests failed. Do you know?

**Comment by cahrens** [comment]
@dvoet this is looking good. Would you mind tossing this branch in a bee so that we can do a sanity check that events are getting sent to the Prometheus endpoint?

**Review by cahrens — APPROVED** [review]

**Review by aherbst-broad — APPROVED** [review]

**Pull request merged**